gtda/homology/simplicial.py

```diff
@@ -532,17 +529,253 @@ def plot(Xt, sample=0, homology_dimensions=None, plotly_params=None):
+            'of': {'type': int, 'in': Interval(0, np.inf, closed='left')}
+            },
+        'coeff': {'type': int, 'in': Interval(2, np.inf, closed='left')},
+        'max_edge_length': {'type': Real},
+        'infinity_values': {'type': (Real, type(None))}
+        }
+
+    def __init__(self, homology_dimensions=(0, 1), coeff=2,
+                 max_edge_length=np.inf, infinity_values=None, n_jobs=None):
+        self.homology_dimensions = homology_dimensions
+        self.coeff = coeff
+        self.max_edge_length = max_edge_length
+        self.infinity_values = infinity_values
+        self.n_jobs = n_jobs
+
+    def _weak_alpha_diagram(self, X):
+        # `indices` will serve as the array of column indices
+        indptr, indices = Delaunay(X).vertex_neighbor_vertices
+
+        # Compute the array of row indices
+        row = np.zeros_like(indices)
+        row[indptr[1:-1]] = 1
+        np.cumsum(row, out=row)
+
+        # We only need the upper diagonal
+        mask = indices > row
+        row, col = row[mask], indices[mask]
+        dists = np.linalg.norm(X[row] - X[col], axis=1)
+        # Note: passing the shape explicitly should not be needed in more
+        # recent versions of C++ ripser
+        n_points = len(X)
+        dm = coo_matrix((dists, (row, col)), shape=(n_points, n_points))
+
+        Xdgms = ripser(dm, maxdim=self._max_homology_dimension,
+                       thresh=self.max_edge_length, coeff=self.coeff,
+                       metric='precomputed')['dgms']
+
+        if 0 in self._homology_dimensions:
+            Xdgms[0] = Xdgms[0][:-1, :]  # Remove one infinite bar
+
+        return Xdgms
+
+    def fit(self, X, y=None):
+        """Calculate :attr:`infinity_values_`. Then, return the estimator.
+
+        This method is here to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray or list
+            Input data representing a collection of point clouds. Can be either
+            a 3D ndarray whose zeroth dimension has size ``n_samples``, or a
+            list containing ``n_samples`` 2D ndarrays.
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        validate_params(
+            self.get_params(), self._hyperparameters, exclude=['n_jobs'])
+        check_point_clouds(X)
+
+        if self.infinity_values is None:
+            self.infinity_values_ = self.max_edge_length
+        else:
+            self.infinity_values_ = self.infinity_values
+
+        self._homology_dimensions = sorted(self.homology_dimensions)
+        self._max_homology_dimension = self._homology_dimensions[-1]
+        return self
+
+    def transform(self, X, y=None):
+        """For each point cloud or distance matrix in `X`, compute the
```

Comment:
Distance matrices are not an allowed input AFAIK.

Reply:
Well spotted, thanks!